holonote/annotate/plotting.py

```diff
@@ -0,0 +1,505 @@
+    opts = _StyleOpts(default={})
+    line_opts = _StyleOpts(default={})
+    span_opts = _StyleOpts(default={})
+    rectangle_opts = _StyleOpts(default={})
+
+    # Editor opts
+    edit_opts = _StyleOpts(default={"line_color": "black"})
+    edit_line_opts = _StyleOpts(default={})
+    edit_span_opts = _StyleOpts(default={})
+    edit_rectangle_opts = _StyleOpts(default={})
+
+    @property
+    def indicator_selection(self) -> dict[str, tuple]:
+        select = {"alpha": (self.selection_alpha, self.alpha)}
+        if self.selection_color is not None:
+            if isinstance(self.color, hv.dim):
+                msg = "'Style.color' cannot be a `hv.dim` when 'Style.selection_color' is not None"
+                raise ValueError(msg)
+            else:
+                select["color"] = (self.selection_color, self.color)
+        return select
+
+    def indicator(self, **select_opts) -> tuple[hv.Options, ...]:
+        opts = {**_default_opts, "color": self.color, **select_opts, **self.opts}
+        return (
+            hv.opts.Rectangles(**opts, **self.rectangle_opts),
+            hv.opts.VSpans(**opts, **self.span_opts),
+            hv.opts.HSpans(**opts, **self.span_opts),
+            hv.opts.VLines(**opts, **self.line_opts),
+            hv.opts.HLines(**opts, **self.line_opts),
+        )
+
+    def editor(self) -> tuple[hv.Options, ...]:
+        opts = {
+            **_default_opts,
+            "alpha": self.edit_alpha,
+            "color": self.edit_color,
+            **self.edit_opts,
+        }
+        return (
+            hv.opts.Rectangles(**opts, **self.edit_rectangle_opts),
+            hv.opts.VSpan(**opts, **self.edit_span_opts),
+            hv.opts.HSpan(**opts, **self.edit_span_opts),
+            hv.opts.VLine(**opts, **self.edit_line_opts),
+            hv.opts.HLine(**opts, **self.edit_line_opts),
+        )
+
+    def reset(self) -> None:
+        params = self.param.objects().items()
+        self.param.update(**{k: v.default for k, v in params if k != "name"})
+
+
+class Indicator:
+    """
+    Collection of class methods that express annotation data as final
+    displayed (vectorized) HoloViews object.
+    """
+
+    @classmethod
+    def points_1d(
+        cls, data, region_labels, fields_labels, invert_axes=False, groupby: str | None = None
+    ):
+        "Vectorizes point regions to VLines. Note does not support hover info"
+        vdims = [*fields_labels, data.index.name]
+        element = hv.VLines(data.reset_index(), kdims=region_labels, vdims=vdims)
+        hover = cls._build_hover_tool(data)
+        return element.opts(tools=[hover])
+
+    @classmethod
+    def points_2d(
+        cls, data, region_labels, fields_labels, invert_axes=False, groupby: str | None = None
+    ):
+        "Vectorizes point regions to VLines * HLines. Note does not support hover info"
+        msg = "2D point regions not supported yet"
+        raise NotImplementedError(msg)
```

Comment:
This wasn't supported because vectorized `HLines` and `VLines` were not available. Should be possible to support now.